sound/soc/sof/sof-audio.c

```diff
@@ -805,6 +805,10 @@ int sof_pcm_stream_free(struct snd_sof_dev *sdev, struct snd_pcm_substream *subs
 	const struct sof_ipc_pcm_ops *pcm_ops = sof_ipc_get_ops(sdev, pcm);
 	int ret;
 
+	/* stop DMA first if needed */
+	if (pcm_ops && pcm_ops->delayed_platform_trigger)
+		snd_sof_pcm_platform_trigger(sdev, substream, SNDRV_PCM_TRIGGER_STOP);
```

Comment:
I don't get this one, this looks exactly the same as in https://github.com/thesofproject/linux/pull/4268/commits/7c22b4fd5d467a1a75aace712a1e5f77eacadc89#diff-e9b5e07876285874ebc72f3a79a32cc92bb8f84a405c8819157d5c954fd0b97aR217, but this time this is not conditional on anything being prepared.

Is this intentional? If yes, a comment to understand the difference would help. This looks like a duplicate or redundant call for now.